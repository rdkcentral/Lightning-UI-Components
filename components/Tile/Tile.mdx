import { Canvas, Story } from '@storybook/addon-docs/blocks';
import { Theme } from '../../.storybook/utils';
import Tile from '.';

# Tile

General purpose Tile component

## Usage

Just set a src property to load an image. From there you can set blur, rounded corners and drop shadow.

```js
import { Tile } from '@lightning/ui';

Tile: {
  type: Tile,
  src: 'path/to/image',
  w: 320,
  h: 180,
  blur: 2,
  radius: 16,
  shadow: { w: 320, h: 180, x: 10, y: 10, zIndex: -1,
    texture: lng.Tools.getShadowRect(320, 180, 8, 4),
    color: 0x00ff00
  }
}
```

<Canvas>
  <Story id="tile--basic" />
</Canvas>

### Properties

name|type|required|default|description
--|--|--|--|--
blur|number|false|-|Blur amount from 0-4
radius|number|false|-|set rounded corner radius
shadow|obj|false|-|create a drop shadow using a Lightning element
src|string|false|-|url to image
fallbackSrc|string|false|-|url to image to use if the requested src fails

### Styles

Styled components can be adjusted by composing using [withStyles](?path=/docs/mixins-withstyles--basic). Below is the default `styles` object

<Theme theme={Tile.styles} />
