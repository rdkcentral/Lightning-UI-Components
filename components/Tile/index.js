import lng from '@lightningjs/core';
import FocusRing from '../FocusRing';
import withStyles from '../../mixins/withStyles';
import blackBackground from '../Styles/black_background_tile';
import { withUpdates, withTags, withHandleKey } from '../../mixins';

export const styles = theme => ({
  radius: theme.border.radius.medium,
  shadow: theme.materials.luminance,
  blur: 0,
  src: blackBackground,
  focusring: function ({ w, h, radius }) {
    return {
      type: FocusRing,
      w,
      h,
      radius: radius + 2,
      size: theme.spacing(0.5),
      zIndex: 1
    };
  },
  unfocused: {
    scale: () => 1,
    focusring: { alpha: 0 },
    shadow: { alpha: 0 }
  },
  focused: {
    scale: theme.getFocusScale,
    focusring: { alpha: 1 },
    shadow: { alpha: 1 }
  }
});

class Tile extends withUpdates(withTags(withHandleKey(lng.Component))) {
  static _template() {
    return {
      Item: {
        Image: {}
      }
    };
  }

  static get tags() {
    return ['Blur', 'DropShadow', 'FocusRing', 'Item', 'Image'];
  }

  static get properties() {
    return ['radius', 'imgRadius', 'blur', 'src', 'fallbackSrc'];
  }

  _construct() {
    this._radius = this.styles.radius;
    this._src = this._fallbackSrc = this.styles.src;
    this._getFocusScale = this.styles.focused.scale;
    this._getUnfocusScale = this.styles.unfocused.scale;
    this._blur = this.styles.blur;
    super._construct && super._construct();
  }

  _init() {
    this._Image.on('txError', () => {
      this.src = this.src !== this.fallbackSrc ? this.fallbackSrc : null;
      this._update();
    });
  }

  _update() {
    this._whenEnabled.then(() => {
      this._Item.w = this.w;
      this._Item.h = this.h;
      this._updateImage();
      this._updateBlur();
      this._updateRadius();
      this._updateDropShadow();
      this._updateFocusRing();
      this._updateScale();
    });
  }

  _updateImage() {
    this._Image.patch({
      rtt: true,
      zIndex: 2,
      w: this.w,
      h: this.h,
      texture: {
        type: lng.textures.ImageTexture,
        src: this._src,
        resizeMode: { type: 'cover', w: this.w, h: this.h }
      }
    });
  }

  _patchBlur() {
    this._Item.patch({
      Blur: {
        type: lng.components.FastBlurComponent,
        w: this.w,
        h: this.h,
        rtt: true,
        zIndex: 3,
        content: {
          Image: {
            w: this._Image.w,
            h: this._Image.h,
            texture: this._Image.getTexture()
          }
        }
      }
    });
  }

  _updateBlur() {
    const alpha = this._blur > 0 ? 1 : 0;
    const amount = this._blur;
    if (this._blur || this._Blur) {
      this._patchBlur();
      if (this._smooth) {
        this._Blur.smooth = { alpha, amount };
      } else {
        this._Blur.patch({ alpha, amount });
      }
    }
  }

  _updateRadius() {
    if (this._radius) {
      if (this._imgRadius === undefined || this._imgRadius !== 0) {
        this._Image.patch({
          shader: {
            type: lng.shaders.RoundedRectangle,
            radius: this._imgRadius || this._radius
          }
        });
      }
      if (this._blur || this._Blur) {
        this._Blur.patch({
          shader: {
            type: lng.shaders.RoundedRectangle,
            radius: this._radius
          }
        });
      }
    }
  }

  _updateDropShadow() {
    if (this.hasFocus() || this._shadow) {
      if (!this._shadow) {
        this._shadow = this.styles.shadow({
          w: this.w,
          h: this.h,
          texture: this._Image.getTexture(),
          radius: this.radius
        });
      }
      let DropShadow = this._shadow;
      const style = this.hasFocus()
        ? this.styles.focused.shadow
        : this.styles.unfocused.shadow;
      if (this._smooth) {
        DropShadow.smooth = style;
      } else {
        DropShadow = { ...DropShadow, ...style };
      }
      this.patch({ DropShadow });
    }
  }

  _updateFocusRing() {
    if (this.hasFocus() || this._focusRing) {
      if (!this._focusRing) {
        this._focusRing = this.styles.focusring({
          w: this.w,
          h: this.h,
          radius: this._radius + 2
        });
      }
      let FocusRingComponent = this._focusRing;
      const style = this.hasFocus()
        ? this.styles.focused.focusring
        : this.styles.unfocused.focusring;
      if (this._smooth) {
        FocusRingComponent.smooth = style;
      } else {
        FocusRingComponent = { ...FocusRingComponent, ...style };
      }
      this.patch({ FocusRing: FocusRingComponent });

      if (this.hasFocus()) {
        this._FocusRing.startAnimation();
      } else {
        this._FocusRing.stopAnimation();
      }
    }
  }

  _updateScale() {
    const scale = this.hasFocus()
      ? this._getFocusScale(this.w)
      : this._getUnfocusScale(this.w);
    if (this._smooth) {
      this._Item.smooth = { scale };
      if (this._FocusRing) this._FocusRing.smooth = { scale };
    } else {
      this._Item.scale = scale;
      if (this._FocusRing) this._FocusRing.scale = scale;
    }
  }

  set shadow(shadow) {
    this._shadow = shadow;
    this._requestUpdateDebounce();
  }

  _focus() {
    if (this._smooth === undefined) this._smooth = true;
    this._update();
  }

  _unfocus() {
    this._update();
  }
}

export default withStyles(Tile, styles);
