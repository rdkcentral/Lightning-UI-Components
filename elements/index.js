export { default as Announcer } from './Announcer';
export { default as Base } from './Base';
export { default as Button } from './Button';
export { default as Icon } from './Icon';
export { default as Keyboard } from './Keyboard';
export { default as MarqueeText } from './MarqueeText';
