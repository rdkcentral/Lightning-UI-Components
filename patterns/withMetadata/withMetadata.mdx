import { Canvas, Story } from '@storybook/addon-docs/blocks';

# withMetadata

The `withMetadata` mixin can be wrapped around any Tile component to position Metadata underneath it. This mixin is not designed for any non-Tile or non-Tile variant components. For example, TileIcon is valid as it extends Tile, but you should not try to use DataItem.

## Usage

Pass in the Tile type to the `withMetadata` mixin and a Metadata component to display with the Tile.

```js
import { withMetadata } from '@lightning/ui';

class Example extends lng.Component {
  static _template() {
    return {
      TileWithMetadata: {
        type: withMetadata(Tile),
        w: 320,
        h: 180,
        persistentMetadata: true,
        Metadata: {
          type: MetadataCard,
          firstLine: 'metadata text here'
        }
      }
    };
  }
};
```

<Canvas>
  <Story id="patterns-mixins-withmetadata" />
</Canvas>

### Properties

name|type|required|default|description
--|--|--|--|--
Metadata|lng.Component|false|-|the component that will be added to the [Tile](?path=/docs/elements-tile--basic), such as [MetadataCard](?path=/docs/elements-metadatacard--basic) on either side or [MetadataTile](?path=/docs/elements-metadatatile--basic) on either side
badge|number|false|-|sets the title text for [Badge](?path=/docs/elements-badge--basic)
badgePadding|number|false|24|sets the amount of space the [Badge](?path=/docs/elements-badge--basic) on the top and left should be inset from the [Tile](?path=/docs/elements-tile--basic) on the top and left
metadataLocation|string|false|'bottom'|where the metadata element will appear relative to the wrapped component. must be one of: `inset` or `bottom`
paddingTop|number|false|8|sets the amount of space between the bottom of the [Tile](?path=/docs/elements-tile--basic) and top of the Metadata
paddingSide|number|false|16|sets the amount of space the Metadata should be inset from the [Tile](?path=/docs/tile--basic) on either side
persistentMetadata|boolean|false|false|determines whether the metadata should only be visible on focus or all the tile
progress|number|false|-|sets the progress percentage of the [ProgressBar](?path=/docs/elements-progressbar--basic)
progressBarPadding|number|false|24|sets the amount of space the [ProgressBar](?path=/docs/elements-progressbar--basic) should be inset from the [Tile](?path=/docs/tile--basic) on the sides and bottom
