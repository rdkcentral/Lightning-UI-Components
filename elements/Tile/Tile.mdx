import { Canvas, Story } from '@storybook/addon-docs';
import { Theme } from '../../.storybook/utils';
import Tile from '.';

# Tile

General purpose Tile component

## Usage

Just set a src property to load an image. From there you can set blur, rounded corners and drop shadow.

```js
import { Tile } from '@lightning/ui';

Tile: {
  type: Tile,
  src: 'path/to/image',
  w: 320,
  h: 180,
  blur: 2,
  radius: 16,
  shadow: { w: 320, h: 180, x: 10, y: 10, zIndex: -1,
    texture: lng.Tools.getShadowRect(320, 180, 8, 4),
    color: 0x00ff00
  }
}
```

<Canvas>
  <Story id="elements-tile--basic" />
</Canvas>

### Properties

| name            | type    | required | default | description                                    |
| --------------- | ------- | -------- | ------- | ---------------------------------------------- |
| blur            | number  | false    | -       | Blur amount from 0-4                           |
| radius          | number  | false    | -       | set rounded corner radius                      |
| imgRadius       | number  | false    | -       | set rounded corner radius of image             |
| shadow          | obj     | false    | -       | create a drop shadow using a Lightning element |
| src             | string  | false    | -       | url to image                                   |
| fallbackSrc     | string  | false    | -       | url to image to use if the requested src fails |
| gradientColor   | color   | false    | -       | overlay gradient color                         |
| focusGradient   | boolean | false    | false   | show gradient overlay on focus                 |
| persistGradient | boolean | false    | false   | always show gradient overlay                   |
| animate         | boolean | false    | false   | animates focus ring                            |

### Styles

Styled components can be adjusted by composing using [withStyles](?path=/docs/mixins-withstyles--basic). Below is the default `styles` object

<Theme theme={Tile.styles} />

### Sizing

Tile can be autosized by composing using [withLayout](?path=/docs/mixins-withlayout--basic).
