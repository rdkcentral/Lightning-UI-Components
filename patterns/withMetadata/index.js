import Tile from '../../elements/Tile';
import withStyles from '../../mixins/withStyles';
import withBadgeProgress from '../withBadgeProgress';

export const styles = theme => ({
  paddingTop: theme.spacing(1),
  paddingSide: theme.spacing(2)
});

export default (base = Tile) =>
  class withMetadata extends withStyles(withBadgeProgress(base), styles) {
    static get properties() {
      return [
        'metadataLocation',
        'paddingSide',
        'paddingTop',
        ...super.properties
      ];
    }

    _construct() {
      this._paddingTop = this.styles.paddingTop;
      this._paddingSide = this.styles.paddingSide;
      this._progressBarPadding = this.styles.progressBarPadding; // available via withBadgeProgress
      this._metadataLocation = this._metadataLocation || 'bottom';
      super._construct();
    }

    _update() {
      super._update();
      this._updateDimensions();
      if (this.Metadata) {
        if (!this.Metadata.zIndex) {
          this.Metadata.zIndex = 5;
        }
        this._updateMetadataAlpha();
        this._updateMetadataW();
        this._updateMetadataH();
        this._updateMetadataX();
        this._updateMetadataY();
        this._updateShowGradient();
      }
    }

    _updateDimensions() {
      super._updateDimensions();
      this._tileScaleOffsetHeight =
        (this._focusedTileHeight - this._unfocusedTileHeight) / 2;
      this._tileScaleOffsetWidth =
        (this._focusedTileWidth - this._unfocusedTileWidth) / 2;
    }

    _updateMetadataAlpha() {
      const unfocusAlpha = this.persistentMetadata ? 1 : 0;
      const alpha = this.hasFocus() ? 1 : unfocusAlpha;
      if (this._smooth) {
        this.Metadata.smooth = { alpha };
      } else {
        this.Metadata.alpha = alpha;
      }
    }

    _updateMetadataH() {
      if (this.metadataLocation === 'inset') {
        if (this._smooth) {
          this.Metadata.smooth = { h: this.h - this.paddingTop };
        } else {
          this.Metadata.h = this.h - this.paddingTop;
        }
      }
    }

    _updateMetadataW() {
      this.Metadata.focusScale =
        this.metadataLocation !== 'inset' ? 1 : this.Metadata.focusScale;

      const currentTileWidth =
        this.hasFocus() || !this.persistentMetadata // prevent logo from sliding horizontally
          ? this._focusedTileWidth
          : this._unfocusedTileWidth;

      this.Metadata.originalW = currentTileWidth - this.paddingSide * 2;
    }

    _updateMetadataX() {
      const focusX = this.paddingSide - this._tileScaleOffsetWidth;
      const unfocusX = this.paddingSide;

      const nextX =
        (this._metadataLocation === 'inset' && !this.persistentMetadata) ||
        this.hasFocus()
          ? focusX
          : unfocusX;

      if (this._smooth) {
        this.Metadata.smooth = { x: nextX };
      } else {
        this.Metadata.x = nextX;
      }
    }

    _updateMetadataY() {
      const focusY = this._calcMetadataLocation();
      const unfocusY = this._metadataLocation === 'inset' ? focusY : focusY;
      this.Metadata.animate =
        !this.persistentMetadata && this.metadataLocation === 'inset';

      const nextY = this.hasFocus() ? focusY : unfocusY;
      if (this._smooth) {
        this.Metadata.smooth = { y: nextY };
      } else {
        this.Metadata.y = nextY;
      }
    }

    _calcMetadataLocation() {
      const currentTileHeight = this.hasFocus()
        ? this._focusedTileHeight
        : this._unfocusedTileHeight;

      const focusOffset = this.hasFocus() ? this._tileScaleOffsetHeight : 0;

      const progressBarOffset = this._ProgressBar
        ? this._ProgressBar.h + this._progressBarPadding
        : 0;

      const metadataHeight = this.Metadata.h;

      const locations = {
        bottom: this.paddingTop + currentTileHeight - focusOffset,
        inset:
          currentTileHeight -
          (metadataHeight +
            progressBarOffset +
            this.paddingTop * 2 +
            focusOffset)
      };

      const metadataY = locations[this._metadataLocation];

      return metadataY;
    }

    _updateShowGradient() {
      if (this.metadataLocation === 'inset') {
        this.focusGradient = true;
      }
      if (this.persistentMetadata && this.metadataLocation === 'inset') {
        this.persistGradient = true;
      }
    }

    _getFocused() {
      return this.Metadata || super._getFocused();
    }

    _setMetadataLocation(metadataLocation) {
      const metadataLocations = ['bottom', 'inset'];
      if (this._metadataLocation !== metadataLocation) {
        if (metadataLocations.includes(metadataLocation)) {
          return metadataLocation;
        } else {
          console.warn(
            `invalid property value: metadataLocation must match one of:${metadataLocations.map(
              Position => ` ${Position}`
            )}`
          );
          return metadataLocations[0];
        }
      }
    }

    set Metadata(Metadata) {
      this.patch({ Metadata });
      this._update();
    }

    get Metadata() {
      return this.tag('Metadata');
    }
  };
