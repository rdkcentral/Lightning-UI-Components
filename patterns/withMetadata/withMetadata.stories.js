import lng from '@lightningjs/core';

import withMetadata from '.';
import mdx from './withMetadata.mdx';
import { Row } from '../../layout';
import {
  MetadataCard,
  MetadataTile,
  TileIcon,
  TileCircle
} from '../../elements';
import lightningbolt from '../../assets/images/ic_lightning_white_32.png';
import pets from '../../assets/images/The_Secret_Life_of_Pets_16x9.jpg';

export default {
  title: 'Patterns/withMetadata',
  parameters: {
    docs: {
      page: mdx
    }
  }
};

export const Basic = args =>
  class Basic extends lng.Component {
    static _template() {
      return {
        TileWithMetadata: {
          type: withMetadata(TileIcon),
          w: 560,
          h: 320,
          persistentMetadata: args.persistentMetadata,
          progress: args.progress,
          badge: args.badge,
          Metadata: {
            type:
              args.metadataType === 'MetadataTile'
                ? MetadataTile
                : MetadataCard,
            firstLine: `${args.metadataType} with a long title to show marquee scroll`,
            secondLine: [
              'and positioned under the tile',
              { badge: 'HD' },
              { badge: 'SD' }
            ],
            logo: lightningbolt,
            logoW: 32
          }
        }
      };
    }

    _getFocused() {
      if (args.focused) {
        return this.tag('TileWithMetadata');
      }
    }
  };
Basic.args = {
  focused: true,
  persistentMetadata: true,
  metadataType: 'MetadataCard',
  badge: 'Live',
  progress: 0.5
};
Basic.argTypes = {
  focused: { control: 'boolean' },
  persistentMetadata: { control: 'boolean' },
  metadataType: {
    control: { type: 'radio', options: ['MetadataCard', 'MetadataTile'] }
  },
  badge: { control: 'text' },
  progress: {
    control: {
      type: 'range',
      min: 0,
      max: 1,
      step: 0.01
    }
  }
};
Basic.parameters = {
  argActions: {
    focused: (isFocused, component) => {
      component._getFocused = isFocused
        ? () => component.tag('TileWithMetadata')
        : () => {};
      component._refocus();
    }
  }
};

export const XfinityTheme = () =>
  class Basic extends lng.Component {
    static _template() {
      return {
        x: 60,
        y: 60,
        Row: {
          type: Row,
          itemSpacing: 60,
          items: [
            {
              type: withMetadata(),
              src: pets,
              w: 320,
              h: 180,
              Metadata: {
                type: MetadataTile,
                firstLine: 'MetadataTile (with long marquee title)',
                secondLine: 'is visible on focus and should truncate'
              }
            },
            {
              type: withMetadata(TileIcon),
              w: 320,
              h: 180,
              icon: lightningbolt,
              iconW: 50,
              iconH: 50,
              persistentMetadata: true,
              progress: 0.5,
              badge: 'HD',
              Metadata: {
                type: MetadataCard,
                firstLine: 'MetadataCard (with long marquee title)',
                secondLine: 'Is always visible',
                thirdLine: '(for focus and unfocus)',
                logo: lightningbolt,
                logoW: 32
              }
            },
            {
              type: withMetadata(TileCircle),
              w: 320,
              h: 180,
              src:
                'https://edge.myriad-gn.top.comcast.net/select/image?entityId=5299164605973043111&default=true&width=320&ratio=3x4&rule=keyArt&quality=100',
              Metadata: {
                type: MetadataTile,
                firstLine: 'Personality entity',
                secondLine: 'with cropped 3x4'
              }
            }
          ]
        }
      };
    }

    _getFocused() {
      return this.tag('Row');
    }
  };
