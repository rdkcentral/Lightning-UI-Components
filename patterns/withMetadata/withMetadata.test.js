import lng from '@lightningjs/core';
import TestUtils from '../../test/lightning-test-utils';
import withMetadata from '.';
import Tile from '../../elements/Tile';
import { MetadataCard, MetadataTile } from '../../elements';

const createTileWithMetadata = (base, props) =>
  TestUtils.makeCreateComponent(withMetadata(base))(props);

describe('withMetadata', () => {
  let tileWithMetadata, testRenderer;
  const w = 320;
  const h = 180;

  beforeEach(() => {
    [tileWithMetadata, testRenderer] = createTileWithMetadata();
  });
  afterEach(() => {
    tileWithMetadata = null;
    testRenderer = null;
  });

  it('renders', () => {
    const tree = testRenderer.toJSON(2);
    expect(tree).toMatchSnapshot();
  });

  it('should update paddingTop and Metadata y position', async done => {
    const paddingTop = 50;
    [tileWithMetadata, testRenderer] = createTileWithMetadata(Tile, {
      w,
      h,
      paddingTop,
      Metadata: { type: MetadataTile }
    });
    await TestUtils.nextTick();
    expect(tileWithMetadata.paddingTop).toBe(paddingTop);
    testRenderer.focus();
    await TestUtils.nextTick();
    expect(tileWithMetadata.Metadata.y).toBe(
      paddingTop +
        tileWithMetadata._focusedTileHeight -
        (tileWithMetadata._focusedTileHeight -
          tileWithMetadata._unfocusedTileHeight) /
          2
    );
    tileWithMetadata._smooth = false;
    testRenderer.unfocus();
    await TestUtils.nextTick();
    expect(tileWithMetadata.Metadata.y).toBe(h + paddingTop);
    done();
  });

  it('should update paddingSide and Metadata x position', async done => {
    const paddingSide = 20;
    [tileWithMetadata, testRenderer] = createTileWithMetadata(Tile, {
      w,
      h,
      paddingSide,
      Metadata: { type: MetadataTile }
    });
    testRenderer.update();
    await TestUtils.nextTick();
    expect(tileWithMetadata.paddingSide).toBe(paddingSide);
    testRenderer.focus();
    await TestUtils.nextTick();
    expect(tileWithMetadata.Metadata.x).toBe(
      paddingSide -
        (tileWithMetadata._focusedTileWidth -
          tileWithMetadata._unfocusedTileWidth) /
          2
    );
    tileWithMetadata._smooth = false;
    testRenderer.unfocus();
    await TestUtils.nextTick();
    expect(tileWithMetadata.Metadata.x).toBe(paddingSide);
    done();
  });

  it('should only display metadata on unfocus if persistentMetadata is true', async done => {
    tileWithMetadata.Metadata = { type: MetadataTile };
    tileWithMetadata._smooth = false;
    testRenderer.update();
    testRenderer.unfocus();
    await TestUtils.nextTick();
    expect(tileWithMetadata.persistentMetadata).toBeUndefined();
    expect(tileWithMetadata.Metadata.alpha).toBe(0.001);
    tileWithMetadata.persistentMetadata = false;
    testRenderer.update();
    testRenderer.unfocus();
    await TestUtils.nextTick();
    expect(tileWithMetadata.persistentMetadata).toBe(false);
    expect(tileWithMetadata.Metadata.alpha).toBe(0.001);
    tileWithMetadata.persistentMetadata = true;
    testRenderer.update();
    testRenderer.unfocus();
    await TestUtils.nextTick();
    expect(tileWithMetadata.persistentMetadata).toBe(true);
    expect(tileWithMetadata.Metadata.alpha).toBe(1);
    done();
  });
});
