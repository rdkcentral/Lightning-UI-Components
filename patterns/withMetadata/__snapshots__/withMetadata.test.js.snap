// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withMetadata renders 1`] = `
Object {
  "Component": Object {
    "FocusRing": Object {
      "active": false,
      "alpha": 1,
      "attached": true,
      "boundsMargin": null,
      "children": Object {
        "Ring": Object {
          "active": false,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": false,
          "flexItem": false,
          "h": 0,
          "isComponent": undefined,
          "mount": 0.5,
          "mountX": 0.5,
          "mountY": 0.5,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Ring",
          "renderOfScreen": undefined,
          "renderToTexture": false,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": Array [
            "Ring",
          ],
          "texture": Object {
            "type": "StaticCanvasTexture",
          },
          "visible": true,
          "w": 0,
          "x": 0,
          "y": 0,
          "zIndex": 0,
        },
      },
      "clipping": false,
      "color": 4294375161,
      "colorBl": "99f6f6f9",
      "colorBr": "26f6f6f9",
      "colorUl": "fff6f6f9",
      "colorUr": "99f6f6f9",
      "enabled": true,
      "flex": false,
      "flexItem": false,
      "h": 0,
      "hasFinalFocus": false,
      "hasFocus": false,
      "isComponent": true,
      "mount": 0,
      "mountX": 0,
      "mountY": 0,
      "pivot": 0.5,
      "pivotX": 0.5,
      "pivotY": 0.5,
      "ref": "FocusRing",
      "renderOfScreen": undefined,
      "renderToTexture": false,
      "scale": 1,
      "scaleX": 1,
      "scaleY": 1,
      "state": "",
      "tag": [Function],
      "tags": Array [
        "FocusRing",
      ],
      "type": "FocusRing",
      "visible": true,
      "w": 0,
      "x": 0,
      "y": 0,
      "zIndex": 10,
    },
    "Item": Object {
      "active": false,
      "alpha": 1,
      "attached": true,
      "boundsMargin": null,
      "children": Object {
        "Image": Object {
          "active": false,
          "alpha": 1,
          "attached": true,
          "boundsMargin": null,
          "clipping": false,
          "color": 4294967295,
          "enabled": true,
          "flex": false,
          "flexItem": false,
          "h": 0,
          "isComponent": undefined,
          "mount": 0,
          "mountX": 0,
          "mountY": 0,
          "pivot": 0.5,
          "pivotX": 0.5,
          "pivotY": 0.5,
          "ref": "Image",
          "renderOfScreen": undefined,
          "renderToTexture": true,
          "scale": 1,
          "scaleX": 1,
          "scaleY": 1,
          "state": undefined,
          "tag": [Function],
          "tags": Array [
            "Image",
          ],
          "texture": Object {
            "type": "ImageTexture",
          },
          "visible": true,
          "w": 0,
          "x": 0,
          "y": 0,
          "zIndex": 2,
        },
      },
      "clipping": false,
      "color": 4294967295,
      "enabled": true,
      "flex": false,
      "flexItem": false,
      "h": 0,
      "isComponent": undefined,
      "mount": 0,
      "mountX": 0,
      "mountY": 0,
      "pivot": 0.5,
      "pivotX": 0.5,
      "pivotY": 0.5,
      "ref": "Item",
      "renderOfScreen": undefined,
      "renderToTexture": false,
      "scale": 1,
      "scaleX": 1,
      "scaleY": 1,
      "state": undefined,
      "tag": [Function],
      "tags": Array [
        "Item",
      ],
      "visible": true,
      "w": 0,
      "x": 0,
      "y": 0,
      "zIndex": 0,
    },
  },
}
`;
